pkgs/build-support/setup-hooks/make-wrapper.sh

```diff
@@ -30,88 +30,99 @@ assertExecutable() {
 makeWrapper() {
     local original="$1"
     local wrapper="$2"
-    local params varName value command separator n fileNames
-    local argv0 flagsBefore flags
+    local -a flagsBefore=()
 
     assertExecutable "$original"
 
     mkdir -p "$(dirname "$wrapper")"
 
-    echo "#! $SHELL -e" > "$wrapper"
+    echo "#! $SHELL -eu" > "$wrapper"
+    echo "declare -a extraFlagsArray=()" >> "$wrapper"
 
-    params=("$@")
-    for ((n = 2; n < ${#params[*]}; n += 1)); do
-        p="${params[$n]}"
-
-        if [[ "$p" == "--set" ]]; then
-            varName="${params[$((n + 1))]}"
-            value="${params[$((n + 2))]}"
-            n=$((n + 2))
+    while (( $# )); do
+    case "$1" in
+        --set)
+            local varName="$2"
```

Comment:
It's not necessary to quote the RHS if it doesn't contain any spaces (even if the variable $2 contains spaces itself)

Reply:
That's true, but based on feedback, I've been doing this sort of quoting for consistency since https://github.com/NixOS/nixpkgs/pull/27614

Reply:
I see, makes sense